getting_started/scripting/gdscript/gdscript_documentation_comments.rst

```diff
@@ -0,0 +1,219 @@
+generated as XML files by the editor.
+
+Documenting a script
+--------------------
+
+Documentation of a script must comes before any other member documentations. The
+script documentation could be divided into three parts.
+
+- A brief description about the script
+- Description of the script
+- Tutorials
+
+To separate these from each other, the documentation comment use special tags
+inside the comments with the tag name followed by an at sign ``@`` and
+immediately before a colon also they must be at the beginning of a line ignoring
+the preceding white spaces.
+
+Tags
+~~~~
+
++-------------------+--------------------------------------------------------+
+| Brief description | No tag and the brief must be at the very beginning of  |
+|                   | the documentation section.                             |
++-------------------+--------------------------------------------------------+
+| Description       | ``@desc:``                                             |
+|                   |                                                        |
++-------------------+--------------------------------------------------------+
+| Tutorial          | ``@tutorial[( The Title Here )]:``                     |
+|                   |                                                        |
++-------------------+--------------------------------------------------------+
+
+**Example:**
+
+::
+
+    extends Node2D
+	
+    ## 
+    ## A brief description about your script.
+    ##
+    ## @desc:
+    ##     The description about the script, what it
+    ##     can do, how to do that and everything.
+    ## 
+    ## @tutorial:            http://the/tutorial1/url.com
+    ## @tutorial(Tutorial2): http://the/tutorial2/url.com
+    ##
+
+.. warning:: If there is any space in between the tag name and colon ``@desc  :``
+             it won't treated as a valid tag and ignored.
+
+
+.. note:: When the description is multi line the preceding and trailing white
+          spaces will be striped and joined with a single space. If the line 
+          break (or any other alignment) is intended use
+          :ref:`bbcode <doc_bbcode_in_richtextlabel>`.
+
+Documenting script members
+--------------------------
+
+Documentation of a script member must immediately precede the member or it's
+annotations if it has any **except for enum values** which's description is should
+be at the the same line as it is. It's designed like this for a better readability
+and look. The description could have more than one line but every line must starts
+with the double hash symbol ``##`` to be considered as it's part of the documentation.
+The script documentation will update in the editor help window every time the
+script updated and if any member variable or function name starts with an
+underscore and undocumented it'll treated as private for the documentation and
+ignored in the in the help window.
```

Comment:
```suggestion
Documentation of a script member must immediately precede the member or its
annotations if it has any. The exception to this is enum values whose description should
be on the same line as the enum for readability. 
The description can have more than one line but every line must start
with the double hash symbol ``##`` to be considered as part of the documentation.
The script documentation will update in the editor help window every time the
script is updated. If any member variable or function name starts with an
underscore it will be treated as private. It will not appear in the documentation and
will be ignored in the help window.
```